### PR TITLE
Add some stricter analysis

### PR DESCRIPTION
**Pull request by natebosch**
Enabled a bunch of lints which are (hopefully) not controversial and fix
existing violations. Most of the fixes are pretty mechanical.

### CODE REVIEW

File: lib/src/io.dart [hunk 1]
```diff
@@ -783,12 +782,12 @@ class _PubProcess {
...
     {String workingDir,
     Map<String, String> environment,
-    bool runInShell: false}) {
+    bool runInShell = false}) {
   // TODO(rnystrom): Should dart:io just handle this?
   // Spawning a process on Windows will not look for the executable in the
   // system path. So, if executable looks like it needs that (i.e. it doesn't
   // have any path separators in it), then spawn it through a shell.
-  if (Platform.isWindows && executable.indexOf('\\') == -1) {
+  if (Platform.isWindows && executable.contains('\\')) {
```
Comment (jakemac53) [inline lib/src/io.dart]: this should be `!executable.contains`

File: lib/src/io.dart [hunk 2]
```diff
@@ -963,10 +962,10 @@ ByteStream createTarGz(List<String> contents, {String baseDir}) {
   return new ByteStream(StreamCompleter.fromFuture(new Future.sync(() async {
     var buffer = new StringBuffer();
     buffer.write('Creating .tar.gz stream containing:\n');
-    contents.forEach((file) => buffer.write('$file\n'));
+    buffer.writeAll(contents, '\n');
```
Comment (jakemac53) [inline lib/src/io.dart]: technically this is not emitting the last newline that would have been emitted before

File: lib/src/io.dart (hunk 1 above)
Comment (natebosch) [reply lib/src/io.dart]: Done

File: lib/src/io.dart (hunk 2 above)
Comment (natebosch) [reply lib/src/io.dart]: Switched to `contents.forEach(buffer.writeln);`

**Pull request merged**